presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    private void closeQuietly(DirContext context)
+    {
+        try {
+            if (context != null) {
+                context.close();
+            }
+        }
+        catch (NamingException ex) {
+            LOG.error("Exception during Context.close()", ex);
+        }
+    }
+
+    private InitialDirContext authenticate(Hashtable<String, String> environment)
+            throws NamingException
+    {
+        return new InitialDirContext(environment);
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain nextFilter)
+            throws IOException, ServletException
+    {
+        // skip auth for http
+        if (!servletRequest.isSecure()) {
+            LOG.debug("Skipping LDAP authentication in HTTP.");
+            nextFilter.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (header != null) {
+            List<String> parts = Splitter.onPattern("\\s+").splitToList(header);
+
+            checkState(parts.size() == 2, "Invalid value for authorization header");
+            checkState(parts.get(0).equalsIgnoreCase(AUTHENTICATION_TYPE), "Incorrect authentication type. Should be basic");
+
+            List<String> credentials = Splitter.on(":").splitToList(new String(Base64.getDecoder().decode(parts.get(1))));
+            checkState(credentials.size() == 2, "Username/Password missing in the request header");
+
+            DirContext context = null;
+            try {
+                String user = credentials.get(0);
+                String password = credentials.get(1);
+
+                Hashtable<String, String> environment = getBasicEnvironment();
+                String principal = getBindDistinguishedName(user);
+
+                environment.put(SECURITY_AUTHENTICATION, "simple");
+                environment.put(SECURITY_PRINCIPAL, principal);
+                environment.put(SECURITY_CREDENTIALS, password);
+                context = authenticate(environment);
+
+                //TODO: Use the context to do the group membership search
+
+                // ldap authentication ok, continue
+                nextFilter.doFilter(new HttpServletRequestWrapper(request)
+                {
+                    @Override
+                    public Principal getUserPrincipal()
+                    {
+                        return new LdapPrincipal(user);
+                    }
+                }, servletResponse);
+                return;
+            }
+            catch (AuthenticationException e) {
+                // LOG incorrect credentials message separately since the one from e is not explicit
+                LOG.debug("Invalid credentials. Check your username/password");
```

Comment:
This message is phrased as though it will be displayed to the end user, but that doesn't make sense because this is going into the server log. How about "LDAP authentication failure due to invalid credentials"